hcache-lmdb.c

```diff
@@ -108,9 +109,18 @@ hcache_lmdb_open(const char *path)
     return NULL;
   }
 
-  mdb_env_set_mapsize(ctx->env, LMDB_DB_SIZE);
+  rc = ENOMEM;
+  while (LMDB_DB_SIZE)
+  {
+      mdb_env_set_mapsize(ctx->env, LMDB_DB_SIZE);
+      rc = mdb_env_open(ctx->env, path, MDB_NOSUBDIR, 0644);
```

Comment:
The LMDB docs for `mdb_env_open` suggest it will return E error numbers on failure and lists a couple (but not ENOMEM).  I might have some time, tonight, to do a little debugging (note neither @gahr nor I can reproduce the problem).

Let us know if you discover any other clues.